### PR TITLE
Add compiler packages to list of packages in Stackage LTS

### PR DESCRIPTION
**Pull request by serras**
As described in https://github.com/input-output-hk/haskell.nix/issues/481, the import logic for the new Stackage format does not include the "compiler packages" as provided by the snapshot. This PR fixes that logic by including those packages with their default revisions.

### CODE REVIEW

File: lts2nix/Main.hs [hunk 1]
```diff
@@ -76,3 +78,5 @@ lts2plan compilerPackagesMap lts = Plan { packages, compilerVersion, compilerPac
             _               -> Nothing
         , packageFlags = Map.lookupDefault Map.empty name flags
         })
+
+    packages = packages' `Map.union` compilerPackages'
```
Comment (angerman) [inline lts2nix/Main.hs]: I think we want the union the other way around. If packages provides overrides for the compilerPackages, we want them to override those, not rely on the compilerPackages.

Comment (serras) [reply lts2nix/Main.hs]: This was my idea too. According to the [docs](http://hackage.haskell.org/package/unordered-containers-0.2.10.0/docs/Data-HashMap-Strict.html#v:union):

> The union of two maps. If a key occurs in both maps, the mapping from the first will be the mapping in the result.

so `packages'` has preference over `compilerPackages'`.

**Comment by serras** [comment]
By the way, it would be great if the Nix expressions for LTS 15.x in the `stackage.nix` repo could be fixed, since right now you cannot use them if you depend on any compiler package.

File: lts2nix/Main.hs (hunk 1 above)
Comment (angerman) [reply lts2nix/Main.hs]: Ohh... that's somewhat counter intuitive... thanks!

**Comment by angerman** [comment]
> By the way, it would be great if the Nix expressions for LTS 15.x in the `stackage.nix` repo could be fixed, since right now you cannot use them if you depend on any compiler package.

Yep. This will need some wrangling, but certainly needs doing!

**Pull request merged**